meshnet/monitor_event_handler.go

```diff
@@ -0,0 +1,161 @@
+package meshnet
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/NordSecurity/nordvpn-linux/config"
+	"github.com/NordSecurity/nordvpn-linux/core/mesh"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+)
+
+var ErrIncorrectCmdlineContent = errors.New("invalid content of cmdline file of /proc")
+
+// FilesharePortAccessController blocks or allows fileshare port when
+// fileshare process stopped or was restarted accordingly.
+type FilesharePortAccessController struct {
+	cm             config.Manager
+	netw           Networker
+	reg            mesh.Registry
+	filesharePID   PID
+	processChecker processChecker
+	mu             sync.Mutex
+}
+
+func NewPortAccessController(cm config.Manager, netw Networker, reg mesh.Registry) FilesharePortAccessController {
+	return FilesharePortAccessController{
+		cm:             cm,
+		netw:           netw,
+		reg:            reg,
+		filesharePID:   0,
+		processChecker: defaultProcChecker{},
+	}
+}
+
+func (eventHandler *FilesharePortAccessController) OnProcessStarted(ev ProcEvent) {
+	if eventHandler.filesharePID != 0 {
+		// fileshare already started and we noted the PID, no need to
+		// process next events anymore until the PID gets reset in [EventHandler.OnProcessStopped]
+		return
+	}
+	if !eventHandler.processChecker.isFileshareProcess(ev.PID) {
+		return
+	}
+	eventHandler.filesharePID = ev.PID
+	go eventHandler.allowFileshare()
```

Comment:
can't there be a problem if multiple routines are starting in the "same" time? which will be the last, here or the `blockFileshare`?